### PR TITLE
editable devilstable

### PR DESCRIPTION
**Pull request by flicksolutions**
I added a [Input-Chip Element](https://www.skeleton.dev/components/input-chips) to the devilstable in order to enhance the functionality while still keeping the table as simple as possible.

### CODE REVIEW

File: src/routes/Devilstable.svelte
Comment (vvvyyynet) [inline src/routes/Devilstable.svelte:151]: hardcoding 'Fassung' is dangerous. Currently renaming 'Fassung' requires 3 edits in Details.svelte and Devilstable.svelte which is too much.

File: src/routes/Devilstable.svelte
Comment (vvvyyynet) [inline src/routes/Devilstable.svelte:107]: consider renaming 'popupClick' to something more specific 'popupChips' or 'popupChipsOnClick'

File: src/routes/Devilstable.svelte
Comment (vvvyyynet) [inline src/routes/Devilstable.svelte:120]: `placement: 'top'` seems to be ignored?!
Furthermore, I noticed that the popup can get hidden behind the topNavBar since the whole section is extending to the top of the viewport... consider adding a margin-top (maybe padding is enough?) to the main content div.

File: src/routes/Devilstable.svelte
Comment (vvvyyynet) [inline src/routes/Devilstable.svelte:109]: "Sie" gross
"einfach" streichen

File: src/routes/Devilstable.svelte
Comment (vvvyyynet) [inline src/routes/Devilstable.svelte:136]: Vorschlag: "Ansicht zurücksetzen"

File: src/routes/Devilstable.svelte
Comment (vvvyyynet) [inline src/routes/Devilstable.svelte:144]: Vorschlag: "Alle entfernen"

File: src/routes/Devilstable.svelte
Comment (vvvyyynet) [inline src/routes/Devilstable.svelte:141]: Consider keeping "Fassung".

File: src/routes/Devilstable.svelte
Comment (vvvyyynet) [inline src/routes/Devilstable.svelte:69]: isn't `$derived([...array])` the same as just `$derived(array)`?

File: src/routes/Devilstable.svelte
Comment (vvvyyynet) [inline src/routes/Devilstable.svelte:70]: Consider renaming `d` and `c` (since `c` is referring to `data` and `d` to `chip`, I got confused... actually just flipping the letters would be perfect!).

Comment (vvvyyynet) [reply src/routes/Devilstable.svelte:70]: Also consider renaming `info`. I just asked ChatGPT and it suggested me `item` or `entry` as singular-forms of `data`. 

ChatGPT-query: `I have an object array called 'data'. How would I name just one element of it? Since there is no singular/plural of data, it's not that easy. Are there conventions?`

File: src/routes/Devilstable.svelte
Comment (vvvyyynet) [inline src/routes/Devilstable.svelte:87]: Is this for 'Fassung' (since all-true)? But why the label 'd'?
If so, consider making 'Fassung' explicit.

File: src/routes/Devilstable.svelte
Comment (vvvyyynet) [inline src/routes/Devilstable.svelte:74]: For better readability (and performance?) consider using `array.fill(false, start, end);` instead of for-loop (maybe +/- 1 correction needed)

File: src/routes/Devilstable.svelte
Comment (vvvyyynet) [inline src/routes/Devilstable.svelte:81]: why 'd'?

File: src/routes/Devilstable.svelte
Comment (vvvyyynet) [inline src/routes/Devilstable.svelte:71]: for better readability consider checking for `d === 'fr'` (and `d === 'Fassung') first and then introduce the info inside the else (or default if switch-statement). Like so, `info` is not (mis?)used for two purposes (1) a switch (relying on it's nullishness if undefined) and (2) as an object array.

File: src/routes/Devilstable.svelte
Comment (vvvyyynet) [inline src/routes/Devilstable.svelte:126]: camelCase?

File: src/routes/Devilstable.svelte
Comment (vvvyyynet) [reply src/routes/Devilstable.svelte:151]: any chance to add this to `sigla.json`, just to have the outsourced?

**Comment by vvvyyynet** [comment]
##### General
- I did not review the types

##### Bugs
- when zooming out the y-axis of Brush is scaling differently than rest of Brush
- when changing size/position of Brush window Detail won't live-update if window-height is longer than 250.
- remove `console.log('scrolling to', targetverse);` in TextzeugenContent.svelte

##### Feature Requests
- make Detail scrollable 
    - scrollListener that changes start/end
    - to update window on Brush, make data_start bindable
    - make sure that on mobile there is enough space left or right of Detail to directly scroll the page
- reduce height of Devilstable such that chips and bottom of the table both fit on screen

##### Documentation
- why is brushE called brushE? (did not expect start/end to be hidden in there)

**Review by vvvyyynet — CHANGES_REQUESTED** [review]
### General
I did not review the types

### Bugs
- when zooming out the y-axis of Brush is scaling differently than rest of Brush
- when changing size/position of Brush window Detail won't live-update if window-height is longer than 250.
- remove console.log('scrolling to', targetverse); in TextzeugenContent.svelte

### Feature Requests
- make Detail scrollable
    - scrollListener that changes start/end to update window on Brush, make data_start bindable
    - make sure that on mobile there is enough space left or right of Detail to directly scroll the page
    - reduce height of Devilstable such that chips and bottom of the table both fit on screen
    
### Documentation
- why is brushE called brushE? (did not expect start/end to be hidden in there)

File: src/routes/Devilstable.svelte
Comment (pdaengeli) [reply src/routes/Devilstable.svelte:70]: > Also consider renaming `info` to make it more explicit. I just asked ChatGPT and it suggested me `item` or `entry` as singular-forms of `data`. (Maybe prepend a `data`, e.g. `data_item`)
> 
> ChatGPT-query: `I have an object array called 'data'. How would I name just one element of it? Since there is no singular/plural of data, it's not that easy. Are there conventions?`

There is always [datum](https://en.wiktionary.org/wiki/datum) but especially in a German-speaking context it may lead to confusion (plus it is marked *dated* in the first sense given).

Comment (vvvyyynet) [reply src/routes/Devilstable.svelte:70]: > There is always datum

Oh, that's nice, but I agree with the caveat you make. Another solution would be to find another word for 'data' alltogether, since it is very generic, and honestly, I had to prop-drill it all the way up to the server.js file to find out what it contains.

**Comment by flicksolutions** [comment]
Thanks so much for your review @vvvyyynet ! As time is of the essence (I want to showcase this tomorrow and on Wednesday), I will review your review now but will **not** wait for your approval.
The normal next steps would be: I react to your comments, you look at my reaction and review again. this would go on until the PR gets approved by you. We will skip this for this PR, but I hope we can do it in the future. :-)

File: src/routes/Devilstable.svelte
Comment (flicksolutions) [reply src/routes/Devilstable.svelte]: Adding it to the sigla-meta-information wouldn't help. (I deleted the sigla.json present in the $lib. It was a dummy file.

I abstracted the string into a constant in constants.js.
Is that ok like that? See changes in aed9718879abff904ea0ecd35d0c4581b44fa357

Comment (vvvyyynet) [reply src/routes/Devilstable.svelte]: oh, I didn't notice it's a dummy file, I found it via global search. 
I'm happy with your solution, `constants.js` is an excellent place.

**Comment by vvvyyynet** [comment]
> As time is of the essence ...
Of course, this makes sense. I could have added to my review that I did not spot any breaking things in your code, just some things that may break when maintaining the code later on. And there are some suggestions for the text, you may consider having a look at today (especially the two buttons).

> The normal next steps would be...
Thanks for explaining this. Let's do this. Also, if you have any stylistic comments of the sort "how to write a good review", please tell me. E.g. for certain comments I was wondering, whether to add them to the end of a block or at the beginning. I have to pick one single line, right?

File: src/routes/Devilstable.svelte
Comment (flicksolutions) [reply src/routes/Devilstable.svelte:82]: there are performance issues at the moment, however I don't think the main culprit lies here...
Created a new issue.

File: src/routes/Devilstable.svelte
Comment (vvvyyynet) [reply src/routes/Devilstable.svelte:24]: Relevant for #30

File: src/routes/Devilstable.svelte
Comment (flicksolutions) [reply src/routes/Devilstable.svelte]: It's not ignored. It is just a _preference_. If there is no space at the top, it will be placed elsewhere.
A simple margin is not possible (since the position of the popup is calculated...
I forced the position to top as a solution. see 5e968978b2313a4467ff3483ae547890ecc1a7fe

File: src/routes/Devilstable.svelte
Comment (flicksolutions) [reply src/routes/Devilstable.svelte:123]: I think the button text gets too long like if I do this.

File: src/routes/Devilstable.svelte
Comment (flicksolutions) [reply src/routes/Devilstable.svelte]: I think the naming of the variable prevents a future coder from just changing this value when they want the default to deviate from the allowed strings. I'm skipping this suggestion because I thinks it's slightly more simplistic. But I really like the suggestion anyways because it shows an eye for future caveats an traps! great comment!

File: src/routes/Devilstable.svelte
Comment (flicksolutions) [reply src/routes/Devilstable.svelte:22]: but these are not the labels! this is the data! it is just filtered by the data

File: src/routes/Devilstable.svelte
Comment (flicksolutions) [reply src/routes/Devilstable.svelte:51]: good catch! :D

File: src/routes/Devilstable.svelte
Comment (flicksolutions) [reply src/routes/Devilstable.svelte:68]: no, just for 'Fassung'. good suggestion!
added it in a comment.

File: src/routes/Devilstable.svelte
Comment (vvvyyynet) [reply src/routes/Devilstable.svelte]: > It's not ignored

Ah, now I understand the placement. So honestly, I'd leave it 

> A simple margin is not possible (since the position of the popup is calculated...

What I meant is adding a margin-top to the great-great-grandparent of the popup, such that it is not hiding behind the top bar. I think, if you then also adjust its **z-index** to the same value as the navbar, then the popup won't hide anymore.

File: src/routes/Devilstable.svelte
Comment (vvvyyynet) [reply src/routes/Devilstable.svelte:123]: I see. Well, you could have them side-by-side and a bit smaller below the input field, but that's not really important now. Happy that you changed the other button.

**However** I'd go for "Zurücksetzen" with capital "Z" (see e.g. the one-word-buttons in github)

File: src/routes/Devilstable.svelte
Comment (vvvyyynet) [reply src/routes/Devilstable.svelte:21]: Oh sorry, my bad. I thought it's a data.map(). Didn't see you are just filtering.

File: src/routes/Devilstable.svelte
Comment (flicksolutions) [reply src/routes/Devilstable.svelte:113]: this is a Skeletoncomponent. I have no influence over the casing of props.

File: src/routes/Devilstable.svelte
Comment (vvvyyynet) [reply src/routes/Devilstable.svelte]: I see your point, but still I'd prefer explicitness over simplicity in this case. Another benefit of the suggested change would be that you could omit introducing `allowStringsWithoutFragments` alltogether and just write everything into `whitelist` (racist terminology by the way, but I know it's baked into Skeleton). The only reason for having a `allowStringsWithoutFragments` is the default values...

Comment (flicksolutions) [reply src/routes/Devilstable.svelte]: that convinced me! :-)

File: src/routes/Devilstable.svelte
Comment (vvvyyynet) [reply src/routes/Devilstable.svelte:24]: You are using `allFractionData` only once in 
```
if (c === 'fr') {
  return allFractionData;
}
```
so why does it need to be empty?

If you need `allFractionData` to be empty when `inputChipValueLabels` does not contain `fr`, then I would
1. drop the guard-clause `if (!inputChipValueLabels.includes('fr')) return {};` 
2. replace `let` by `const`
3. move the logic outside the definition, like so:

```
allFractionDataWhichCanBeEmpty = inputChipValueLabels.includes('fr')) ? allFractionData : {}
```

File: src/routes/Devilstable.svelte
Comment (vvvyyynet) [reply src/routes/Devilstable.svelte]: oh, this is not intuitive to me. Hearing `label` I think of an axis-label or a popup-label, so something that is written-out and hence also human readable.

So what is the difference between `label` and `handle` in the example above? I think both are "mk", right?

File: src/routes/Devilstable.svelte
Comment (vvvyyynet) [reply src/routes/Devilstable.svelte:116]: I meant `name="Inputchips"` --> `name="inputChips"`. I think you can choose any value for name, can't you?!

**Pull request merged**
